tools/settings.py

```diff
@@ -156,6 +129,16 @@ def limit_settings(self, allowed):
     if allowed:
       self.allowed_settings.update(allowed)
 
+  def update_port_settings(self, port_setting):
+    for key, value in port_setting.items():
+      default = self.attrs.setdefault(key, value)
+      if value != default:
+        raise ValueError(
+          f"port setting {key!r}: '{value!r}' != default '{default!r}'."
+        )
```

Comment:
Would it be possible to instead just remove the settings from the settings.js file?   Then this check would presumably be not longer needed?

Reply:
What is the `!r` syntax doing there?

Reply:
`!r` uses `__repr__` instead of `__str__`.